### PR TITLE
Prep Javascript Loops

### PR DESCRIPTION
**Pull request by Garcia-Tony**
Screen Shot 2024-04-29 at 8 07 21 PM
Screen Shot 2024-04-29 at 8 07 48 PM
Screen Shot 2024-04-29 at 8 08 17 PM
Screen Shot 2024-04-29 at 8 08 44 PM
Screen Shot 2024-04-29 at 8 09 26 PM

### CODE REVIEW

File: prep-javascript-loops/main.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+// create your loops here.
+function whileLoop1() {
+  const numArray = [0, 1, 2, 3, 4, 5, 6, 7, 8];
```
Comment (BrettAlbright) [inline prep-javascript-loops/main.js]: this array should be empty when its defined. 
The loop should be populating it.

File: prep-javascript-loops/main.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  let index = 0;
+  while (index < 9) index++;
```
Comment (BrettAlbright) [inline prep-javascript-loops/main.js]: this loop is not currently working. 

Make sure to retest after clearing out the array above.

File: prep-javascript-loops/main.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  numArray.push(index);
+  return numArray;
+}
+
+const whileLoop1Result = whileLoop1(9);
+console.log('whileLoop1Result', whileLoop1Result);
+
+function whileLoop2() {
+  const numArray = [0, 2, 4, 6, 8, 10, 12, 14, 16];
```
Comment (BrettAlbright) [inline prep-javascript-loops/main.js]: same as above, this array should be empty when it is defined.

File: prep-javascript-loops/main.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  let index = 0;
+  while (index < 18) index++;
```
Comment (BrettAlbright) [inline prep-javascript-loops/main.js]: this loop will need to be updated.

File: prep-javascript-loops/main.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  numArray.push(index);
+  return numArray;
+}
+
+const whileLoop2Result = whileLoop2(18);
+console.log('whileLoop2Result', whileLoop2Result);
+
+function forLoop1() {
+  const numArray = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
```
Comment (BrettAlbright) [inline prep-javascript-loops/main.js]: same as above, this array should start empty

File: prep-javascript-loops/main.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  for (let index = 0; index < 10; index++) return numArray;
```
Comment (BrettAlbright) [inline prep-javascript-loops/main.js]: This loop should be used to populate the numArray.

File: prep-javascript-loops/main.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+console.log('forLoop1Result', forLoop1Result);
+
+function forLoop2() {
+  for (let index = 101 - 1; index >= 0; index--) {
+    const stringVal = 'Time to Explosion: ';
+    console.log('Time to Explosion' + ' ' + index);
+  }
+}
+
+var forLoops2 = forLoop2('stringVal' + 101);
```
Comment (BrettAlbright) [inline prep-javascript-loops/main.js]: this function should not be called with an argument.

**Review by BrettAlbright — CHANGES_REQUESTED** [review]
Some updates are required. 

Make sure that when you create a new array in the function, you define an empty array and then populate it using the loop.

Reach out if you have questions!

**Comment by Garcia-Tony** [comment]
Screen Shot 2024-04-30 at 7 46 06 PM

**Review by BrettAlbright — APPROVED** [review]
👍🏼

**Pull request merged**